pbs/pbsrequest.go

```diff
@@ -30,20 +37,42 @@ type Bids struct {
 	Params     json.RawMessage `json:"params"`
 }
 
+type PBSVideo struct {
+	Mimes          []string `json:"mimes,omitempty"`
+	Minduration    uint64   `json:"minduration,omitempty"`
+	Maxduration    uint64   `json:"maxduration,omitempty"`
+	Startdelay     int64    `json:"startdelay,omitempty"`
+	Skippable      int      `json:"skippable,omitempty"`
+	PlaybackMethod int8     `json:"playback_method,omitempty"`
+	Frameworks     []string `json:"frameworks,omitempty"`
+}
+
 type AdUnit struct {
-	Code     string           `json:"code"`
-	TopFrame int8             `json:"is_top_frame"`
-	Sizes    []openrtb.Format `json:"sizes"`
-	Bids     []Bids           `json:"bids"`
-	ConfigID string           `json:"config_id"`
+	Code       string           `json:"code"`
+	TopFrame   int8             `json:"is_top_frame"`
+	Sizes      []openrtb.Format `json:"sizes"`
+	Bids       []Bids           `json:"bids"`
+	ConfigID   string           `json:"config_id"`
+	MediaTypes []string         `json:"media_types"`
 }
 
 type PBSAdUnit struct {
-	Sizes    []openrtb.Format
-	TopFrame int8
-	Code     string
-	BidID    string
-	Params   json.RawMessage
+	Sizes      []openrtb.Format
+	TopFrame   int8
+	Code       string
+	BidID      string
+	Params     json.RawMessage
+	Video      PBSVideo
+	MediaTypes []MediaType
```

Comment:
I'm not sure :/. That's actually the main reason I suggested changing `PBSAdUnit` instead. A Function call is like two people talking, and the Arguments are the words they exchange. The struct defines the Language... and if it allows for unclear speech, then the inevitable Misunderstandings become future bugs.

If that's still not convincing, though... I'd lean towards "error."

If a request with this data exists, then it's almost certainly a bug in _someone's_ code. If I were debugging, I'd much rather get an error like "Video property is required when MediaType contains video" than trying to find out why I keep getting back empty video data.